homeassistant/components/homekit_controller/__init__.py

```diff
@@ -100,9 +123,10 @@ def _setup_characteristic(self, char):
         # pylint: disable=not-callable
         setup_fn(char)
 
-    def handle_new_values(self, new_values):
-        """Update state from one of more characteristics fetch from device."""
-        for (_, iid), result in new_values.items():
+    async def async_state_changed(self):
```

Comment:
Make this a callback instead of a coroutine, since we don't need to await inside. Decorate it with `@callback` imported from core.py.

Reply:
Done